src/index.ts

```diff
@@ -121,118 +150,134 @@ const makeContainer = ({
     buttonsDiv.appendChild(nextButton);
 
     container.appendChild(buttonsDiv);
+  }
+  // autoplay
+  if (playing) {
+    playingTimer = setInterval(() => {
+      increment();
+    }, interval);
+  }
 
-    return [container, titleDiv, slider];
+  return [container, titleDiv, slider];
 };
 
-type Position = 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right';
 
 type TemporalFrame = {
-    title: string;
-    layers: AnyLayer[];
+  title: string;
+  layers: LayerSpecification[];
 };
 
 type Options = {
-    position?: Position;
-    interval?: number;
-    performance?: boolean;
+  position?: ControlPosition;
+  interval?: number;
+  loopDelay?: number;
+  performance?: boolean;
+  showButtons?: boolean;
+  loop?: boolean;
+  autoplay?: boolean;
 };
 
 export default class TemporalControl implements IControl {
-    private map: Map | undefined;
-    private options: Options;
-
-    private container: HTMLDivElement;
-    private containerTitle!: HTMLDivElement;
-    private temporalSlider!: HTMLInputElement;
-    private temporalFrames: TemporalFrame[];
-
-    constructor(temporalFrames: TemporalFrame[], options: Options = {}) {
-        this.temporalFrames = temporalFrames;
-        this.options = options;
-
-        const containerOptions: ContainerOptions = {
-            length: this.temporalFrames.length,
-            interval: this.options.interval || 500,
-            onSliderValueChange: () => this.refresh(),
-        };
-
-        [this.container, this.containerTitle, this.temporalSlider] =
-            makeContainer(containerOptions);
-    }
+  private map: Map | undefined;
+  private options: Options;
 
-    onAdd(map: Map) {
-        this.map = map;
-        map.getContainer().appendChild(this.container);
+  private container: HTMLDivElement;
+  private containerTitle!: HTMLDivElement;
+  private temporalSlider!: HTMLInputElement;
+  private temporalFrames: TemporalFrame[];
 
-        this.map.once('styledata', () => {
-            this.refresh();
-        });
+  constructor(temporalFrames: TemporalFrame[], options: Options = {}) {
+    this.temporalFrames = temporalFrames;
+    this.options = options;
 
-        return this.container;
-    }
+    const containerOptions: ContainerOptions = {
+      length: this.temporalFrames.length,
+      interval: this.options.interval || 500,
+      loopDelay: this.options.loopDelay || 1000,
+      showButtons: this.options.showButtons || false,
```

Comment:
I want `showButtons` to be true as default.

```suggestion
      showButtons: this.options.showButtons || true,
```